src/core/functions/table/match.cpp

```diff
@@ -1138,10 +1203,32 @@ PGQMatchFunction::MatchBindReplace(ClientContext &context,
       continue;
     }
 
-    // TODO(hjiang): For star expression, only select columns in vertex or edge
-    // table, but not those unspecified in property graph.
-    // Issue reference: https://github.com/cwida/duckpgq-extension/issues/192
-    final_column_list.push_back(std::move(expression));
+    // Handle StarExpression.
+    auto *star_expression = dynamic_cast<StarExpression *>(expression.get());
+    if (star_expression != nullptr) {
+      auto selected_col_names =
+          star_expression->relation_name.empty()
+              ? GetRegisteredColFromPg(alias_to_vertex_and_edge_tables)
+              : GetRegisteredColFromPg(alias_to_vertex_and_edge_tables,
+                                       star_expression->relation_name);
+
+      // Fallback to star expression if cannot figure out the columns to query.
+      if (selected_col_names.empty()) {
+        final_column_list.emplace_back(std::move(expression));
+        continue;
+      }
+
+      final_column_list.reserve(final_column_list.size() +
+                                selected_col_names.size());
+      for (auto &col : selected_col_names) {
```

Comment:
If we make the functions return ColumnRefExpressions, this bit can also be simplified